extensions/amp-analytics/0.1/events.js

```diff
@@ -32,6 +33,120 @@ const VARIABLE_DATA_ATTRIBUTE_KEY = /^vars(.+)/;
 const NO_UNLISTEN = function() {};
 const TAG = 'analytics-events';
 
+/**
+ * Events that can result in analytics data to be sent.
+ * @const
+ * @enum {string}
+ */
+export const AnalyticsEventType = {
+  VISIBLE: 'visible',
+  CLICK: 'click',
+  TIMER: 'timer',
+  SCROLL: 'scroll',
+  HIDDEN: 'hidden',
+};
+
+const ALLOWED_FOR_ALL_ROOT_TYPES = ['ampdoc', 'embed'];
+
+/**
+ * Events that can result in analytics data to be sent.
+ * @const {!Object<string, {
+ *     name: string,
+ *     allowedFor: !Array<string>,
+ *     klass: function(new:./events.EventTracker)
+ *   }>}
+ */
+const TRACKER_TYPE = Object.freeze({
+  'click': {
+    name: 'click',
+    allowedFor: ALLOWED_FOR_ALL_ROOT_TYPES.concat(['timer']),
+    // Escape the temporal dead zone by not referencing a class directly.
+    klass: function(root) { return new ClickEventTracker(root); },
```

Comment:
Why indirect the constructor?

Reply:
This is an ES6 problem, in that class types do not get 'hoisted' (aka moved to the "top" when declared, this problem is known as the temporal dead zone), so when this code is run it actually ends up always being klass = null when we use the class reference directly like 'klass: ClickEventTracker'. The perils of using an no-compilation language, basically.

Reply:
/cc @erwinmombay @jridgewell